sys/kern/link_elf.c

```diff
@@ -627,6 +676,36 @@ parse_dynamic(elf_file_t ef)
 		}
 	}
 
+	/*
+	 * Set bounds on all section pointers for which we know the size
+	 * If we are not a cheri kernel these are no-ops.
+	 */
+	if (ef->strtab != NULL)
+		ef->strtab = cheri_kern_setbounds(ef->strtab, ef->strsz);
+	if (ef->rel != NULL)
+		ef->rel = cheri_kern_setbounds(ef->rel, ef->relsize);
+	if (ef->pltrel != NULL)
+		ef->pltrel = cheri_kern_setbounds(ef->pltrel, ef->pltrelsize);
+	if (ef->rela != NULL)
+		ef->rela = cheri_kern_setbounds(ef->rela, ef->relasize);
+	if (ef->symtab != NULL)
+		ef->symtab = cheri_kern_setbounds(ef->symtab,
+		    ef->nchains * sizeof(Elf_Sym));
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(DT_CHERI___CAPRELOCS)
+	if (ef->caprelocs != NULL)
+		ef->caprelocs = cheri_kern_setbounds(ef->caprelocs,
+		    ef->caprelocssize);
+#endif
+
+	/*
+	 * XXX-AM: How do we get .got size?
+	 * We probably have to leave it like this or define
+	 * a symbol after GOT, in any case the got capability
+	 * would be limited by ef->address capability.
+	 */
+	/* if (ef->got) */
+	/* 	ef->got */
```

Comment:
Honestly? The field is unused, it should be deleted, nothing needs to iterate over it (and nothing uses it AFAICT).

Reply:
I can't see any other use, leaving this open so that I remember to PR upstream.